go-app.js

```diff
@@ -215,14 +218,84 @@ go.app = function() {
     });
 
     self.states.add('Submit_Case', function(name) {
-      return new EndState(name, {
-        text: "Thank you! Your report has been submitted.",
-        next_state: 'Facility_Code_Entry'
-      });
+      return Q()
+        .then(function() {
+          // Send response to Ona
+          var ona_conf = self.im.config.ona;
+          if (typeof ona_conf == 'undefined') {
+            return self.im.log.info([
+              "No Ona API configured.",
+              "Not submitting data to Ona."
+            ].join(" "));
+          }
+          var ona = new Ona(self.im, {
+            auth: {
+              username: ona_conf.username,
+              password: ona_conf.password
+            },
+            url: ona_conf.url
+          });
+          var submission = self.create_ona_submission(data);
+          return ona.submit({
+            id: self.im.config.ona.id,
+            submission: submission,
+          });
+        })
+        .fail(function(err) {
+          // Check for Ona response error
+          return self.im.log.error([
+            'Error when sending data to Ona:',
+            JSON.stringify(err.response)
+          ].join(' '));
+        })
+        .then(function() {
+          return new EndState(name, {
+            text: "Thank you! Your report has been submitted.",
+            next_state: 'Facility_Code_Entry'
+          });
+        });
 
     });
     // END NEW STEPS
 
+    self.create_ona_submission = function(data) {
+      var submission = {
+        facility_code: data.Facility_Code_Entry,
+        reported_by: self.im.user.addr,
+        first_name: data.First_Name_Entry,
+        id_type: data.ID_Type_Entry,
+        last_name: data.Last_Name_Entry,
+        locality: data.Locality_Entry,
+        msisdn: data.MSISDN_Entry,
+        date_of_birth: data.dob,
+        create_date_time: self.now(),
+        abroad: data.Patient_Abroad_Entry,
+        sa_id_number: data.SA_ID_Entry,
+        gender: data.No_SA_ID_Gender_Entry,
+
+      };
+
+      // if ((typeof data.toilet.code === 'string') &&
+      //     (typeof data.toilet.lat === 'number') &&
+      //     (typeof data.toilet.lon === 'number')) {
+      //     var offsets = self.calculate_gps_offsets(data.toilet.code);
+      //     submission.toilet_location = [
+      //         data.toilet.lat + offsets.lat,
+      //         data.toilet.lon + offsets.lon,
+      //     ].join(' ');
+      // }
```

Comment:
You mean the comments?
I wanted it there, because the next phase would be:

uncomment that and make it work.
not have to go back to imali and re-copy that in (which is how i got here in the first place)

Incremental improvements.